x/bank/migrations/v043/json.go

```diff
@@ -0,0 +1,32 @@
+package v043
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// pruneZeroBalancesJSON removes the zero balance addresses from exported genesis.
+func pruneZeroBalancesJSON(oldBalances []types.Balance) []types.Balance {
+	var balances []types.Balance
+
+	for _, b := range oldBalances {
+		if !b.Coins.IsZero() {
+			b.Coins = sdk.NewCoins(b.Coins...) // prunes zero denom.
+			balances = append(balances, b)
```

Comment:
Why modify `b` and not just `balances = append(balances, sdk.NewCoins(b.Coins...))`?

Reply:
> Why modify `b` and not just `balances = append(balances, sdk.NewCoins(b.Coins...))`?

if we do this, we may miss `b.Address`.

Reply:
I don't follow. We don't use `b.Address`. Are `oldBalances` being mutated?

Reply:
yes, we need to remove the some zero denoms from balance of an address (occures when few denoms set to zero but not all)

ex:
```
{
	"address": "cosmos1fl48vsnmsdzcv85q5d2q4z5ajdha8yu34mf0eh",
	"coins": [
		{
			"amount": "10",
			"denom": "foo"
		},
		{
			"amount": "20",
			"denom": "bar"
		},
		{
			"amount": "0",
			"denom": "foobar"
		}
	]
}
```

we have to remove `foobar` here.